velox/functions/prestosql/window/tests/RowNumberTest.cpp

```diff
@@ -0,0 +1,177 @@
+              rowType, size, *pool_, isNullAt));
+      vectors.push_back(vector);
+    }
+    return vectors;
+  }
+
+  std::shared_ptr<const RowType> rowType_{
+      ROW({"c0", "c1", "c2", "c3", "c4", "c5", "c6"},
+          {BIGINT(),
+           SMALLINT(),
+           INTEGER(),
+           BIGINT(),
+           REAL(),
+           DOUBLE(),
+           VARCHAR()})};
+  folly::Random::DefaultGenerator rng_;
+};
+
+TEST_F(RowNumberTest, basicRowNumber) {
+  auto testWindowSql = [&](const RowVectorPtr& input,
+                           const std::string& windowSql) -> void {
+    VELOX_CHECK_GE(input->size(), 2);
+
+    auto op = PlanBuilder()
+                  .values({input})
+                  .project({"c0 as c0", "c1 as c1"})
+                  .window({windowSql})
+                  .orderBy({"c0 asc nulls last", "c1 asc nulls last"}, false)
+                  .planNode();
+    assertQuery(
+        op, "SELECT c0, c1, " + windowSql + " FROM tmp order by c0, c1");
+  };
+
+  auto basicTests = [&](const RowVectorPtr& vectors) -> void {
+    testWindowSql(
+        vectors,
+        "row_number() over (partition by c0 order by c1) as row_number_partition");
+    testWindowSql(
+        vectors,
+        "row_number() over (partition by c1 order by c0) as row_number_partition");
+    testWindowSql(
+        vectors,
+        "row_number() over (partition by c0 order by c1 desc) as row_number_partition");
+    testWindowSql(
+        vectors,
+        "row_number() over (partition by c1 order by c0 desc) as row_number_partition");
+
+    // No partition clause
+    testWindowSql(
+        vectors, "row_number() over (order by c0, c1) as row_number_partition");
+    testWindowSql(
+        vectors, "row_number() over (order by c1, c0) as row_number_partition");
+    testWindowSql(
+        vectors,
+        "row_number() over (order by c0 asc, c1 desc) as row_number_partition");
+    testWindowSql(
+        vectors,
+        "row_number() over (order by c1 asc, c0 desc) as row_number_partition");
+
+    // No order by clause
+    testWindowSql(
+        vectors,
+        "row_number() over (partition by c0, c1) as row_number_partition");
+  };
+  vector_size_t size = 100;
+  auto valueAtC0 = [](auto row) -> int32_t { return row % 5; };
+  auto valueAtC1 = [](auto row) -> int32_t { return row % 7; };
+
+  auto vectors = makeRowVector({
+      makeFlatVector<int32_t>(size, valueAtC0),
+      makeFlatVector<int32_t>(size, valueAtC1),
+  });
+
+  createDuckDbTable({vectors});
+  basicTests(vectors);
+
+  // Test all input in a single partition.
+  size = 50;
+  auto valueAtC01 = [](auto row) -> int32_t { return 1; };
+  auto valueAtC11 = [](auto row) -> int32_t { return row; };
+
+  vectors = makeRowVector({
+      makeFlatVector<int32_t>(size, valueAtC0),
+      makeFlatVector<int32_t>(size, valueAtC1),
+  });
+
+  createDuckDbTable({vectors});
+  basicTests(vectors);
+}
+
+TEST_F(RowNumberTest, rowNumberRandomGen) {
```

Comment:
Do we have test coverage for the cases where partition doesn't fit into a single output batch or when partition boundary is aligned with the output batch or other edge cases like that?